### PR TITLE
feat: support configuring SES for email sender

### PR DESCRIPTION
**Pull request by ShadowCat567**
## Problem

**Issue number, if available:** N/A

## Changes

* exposed parameters to configure Cognito to send emails with SES
* updated auth-construct to be able to feed new properties (related to configuring SES) to User Pool settings
* added corresponding test to validate SES email configuration is set up if user provides data for it

**Corresponding docs PR, if applicable:**

## Validation

- added unit test to verify email configuration happens when subfields within senders (fromEmail, fromName, and replyTo) are populated
- manually deployed test project with SES email configuration included (record of successful deployment below)
successful_deployment_with_SES

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by sobolk — DISMISSED** [review]

**Review by rtpascual — DISMISSED** [review]

File: packages/auth-construct/src/construct.test.ts [hunk 1]
```diff
@@ -545,6 +545,34 @@ void describe('Auth construct', () => {
...
+      senders: {
+        email: sesEmailSettings,
+      },
+    });
+
+    const template = Template.fromStack(stack);
+    template.allResourcesProperties('AWS::Cognito::UserPool', {
+      EmailConfiguration: {
+        From: expectedEmail,
+        ReplyToEmailAddress: expectedReply,
```
Comment (awsluja) [inline packages/auth-construct/src/construct.test.ts:571]: can you validate the fromName value as well?

Comment (ShadowCat567) [reply packages/auth-construct/src/construct.test.ts:571]: yep!

Comment (ShadowCat567) [reply packages/auth-construct/src/construct.test.ts:571]: The From field of EmailConfiguration validates both fromName and fromEmail, the variable expectedEmail is structured "fromName <fromEmail>". In EmailConfiguration, there is not a separate field that is only for fromName or a separate field for fromEmail, attached is the properties related to EmailConfiguration highlighted in red.
email-template
 I can update the variable name to make this more clear.

Comment (awsluja) [reply packages/auth-construct/src/construct.test.ts:571]: oh i see

**Review by awsluja — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**